linq4j/src/main/java/org/apache/calcite/linq4j/tree/ConstantExpression.java

```diff
@@ -118,7 +126,15 @@ private static ExpressionWriter write(ExpressionWriter writer,
         if (value instanceof BigDecimal) {
           bigDecimal = (BigDecimal) value;
         } else {
-          bigDecimal = BigDecimal.valueOf((Double) value);
+          double d = (Double) value;
+          if (d == Double.POSITIVE_INFINITY) {
```

Comment:
I see that you're checking for `Double` and `Float`. I'd like to avoid adding another level of nesting + adding two very similar logic blocks if possible. Do you think a helper could help fix this?

Reply:
It's fine to leave it I am just weary of adding similar nested `if` blocks. If you don't think there is a way to improve it that is ok.